packages/agents/src/agents/Scripter/Scripter.ts

```diff
@@ -0,0 +1,32 @@
+import { LlmApi, Chat, Workspace, Scripts, Env, agentPlugin, WrapClient, Logger } from "@evo-ninja/agent-utils";
+import { AgentBase } from "../../AgentBase";
+import { SCRIPTER_AGENT_CONFIG, ScripterContext, ScripterRunArgs } from "./config";
+
+export class Scripter extends AgentBase<ScripterRunArgs, ScripterContext> {
+  constructor(
+    llm: LlmApi,
+    chat: Chat,
+    logger: Logger,
+    workspace: Workspace,
+    scripts: Scripts,
+    env: Env
+  ) {
+    const agentContext = {
+      llm,
+      chat,
+      workspace,
+      scripts,
+      logger,
+      globals: {},
+      client: new WrapClient(
+        workspace,
+        logger,
+        agentPlugin({ logger }),
```

Comment:
Now that we don't have onGoalAchieved(or failed) or speak, do we still need the agent plugin(in any agent)?

Reply:
We still have `onGoalAchieved` and `onGoalFailed`, but I do think we could revisit the need for having a script + plugin for its implementation. Currently the call path looks like this `llm-function -> script -> plugin-function`, which I think could be simplified into simply `llm-function`.